mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,173 @@
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawSet
+        A Raw object containing EEGLAB .set data.
+    """
+    return RawSet(fname=fname, ch_fname=ch_fname, eog=eog, preload=preload,
+                  verbose=verbose)
+
+
+class RawSet(_BaseRaw):
+    """Raw object from EEGLAB .set file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the .set file.
+    ch_fname : str
+        Path to the file containing channel locations. Currently,
+        only .loc extension is supported.
+    eog : list or tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawSet
+        A Raw object containing EEGLAB .set data.
+    """
+    def __init__(self, fname, ch_fname, eog=None, preload=False, verbose=None):
+        """Read EEGLAB .set file.
+        """
+        basedir = op.dirname(fname)
+        eeg = io.loadmat(fname, struct_as_record=False, squeeze_me=True)['EEG']
+
+        # read the data
+        data_fname = op.join(basedir, eeg.data)
+        logger.info('Reading %s' % data_fname)
+
+        last_samps = [int(eeg.xmax * eeg.srate)]
+
+        # get info
+        if ch_fname is not None:
+            ch_fname = op.join(basedir, ch_fname)
+        info = _get_info(eeg, eog, ch_fname)
+
+        super(RawSet, self).__init__(
+            info, preload, filenames=[data_fname], last_samps=last_samps,
+            orig_format='double', verbose=verbose)
+        logger.info('    Range : %d ... %d =  %9.3f ... %9.3f secs'
+                    % (self.first_samp, self.last_samp,
+                       float(self.first_samp) / self.info['sfreq'],
+                       float(self.last_samp) / self.info['sfreq']))
```

Comment:
You shouldn't need to do this, it should be done by the base class if it isn't already. Did you take this from some other reader? If it was BTi or EGI that's being changed here #2668